src/exchange.ts

```diff
@@ -0,0 +1,131 @@
+import BN from "bn.js";
+import { BatchExchangeInstance } from "../types/truffle-typings";
+
+export interface TokenInfo {
+  id: number;
+  symbol: string;
+  decimals: number;
```

Comment:
Given that they can be UNKNOWN should these fields just be optional?

Reply:
I would be happy with this. Much better then having a fallback!

Reply:
Just want to elaborate a bit here; There are really only two scenarios where decimals are unknown (at least with regards to the intended purpose of this interface). That is, 
1. If there is no token registered at the specified ID on the exchange
2. If the code at the address registered on the exchange is not that of an ERC20.

In both scenarios, this is essentially equivalent to `TokenNotFound` and using `null` for symbol and decimals would make perfect sense!